jujugui/static/gui/src/app/components/user-profile/entity-list.js

```diff
@@ -0,0 +1,351 @@
+            {bundle.owner}
+          </span>
+        </juju.components.UserProfileEntity>);
+    },
+
+    /**
+      Generate the header for the bundles.
+
+      @method _generateBundleHeader
+      @returns {Array} The markup for the header.
+    */
+    _generateBundleHeader: function() {
+      return (
+        <li className="user-profile__list-header twelve-col">
+          <span className="user-profile__list-col five-col">
+            Name
+          </span>
+          <span className={'user-profile__list-col three-col ' +
+            'user-profile__list-icons'}>
+            Charms
+          </span>
+          <span className="user-profile__list-col one-col prepend-one">
+            Units
+          </span>
+          <span className={
+            'user-profile__list-col two-col last-col'}>
+            Owner
+          </span>
+        </li>);
+    },
+
+    /**
+      Generate the details for the provided charm.
+
+      @method _generateCharmRow
+      @param {Object} charm A charm object.
+      @returns {Array} The markup for the row.
+    */
+    _generateCharmRow: function(charm) {
+      var id = charm.id;
+      // Ensure the icon is set.
+      charm.icon = charm.icon || this._getIcon(id);
+      return (
+        <juju.components.UserProfileEntity
+          changeState={this.props.changeState}
+          entity={charm}
+          key={id}
+          type="charm">
+          <span className={'user-profile__list-col three-col ' +
+            'user-profile__list-name'}>
+            {charm.name}
+            {this._generateTags(charm.tags, id)}
+          </span>
+          <span className="user-profile__list-col four-col">
+            {this._generateSeries(charm.series, id)}
+          </span>
+          <span className={'user-profile__list-col one-col ' +
+            'user-profile__list-icons'}>
+            <img className="user-profile__list-icon"
+              src={charm.icon}
+              title={charm.name} />
+          </span>
+          <span className={'user-profile__list-col two-col ' +
+            'prepend-two last-col'}>
+            {charm.owner}
+          </span>
+        </juju.components.UserProfileEntity>);
+    },
+
+    /**
+      Generate the header for the charms.
+
+      @method _generateCharmHeader
+      @returns {Array} The markup for the header.
+    */
+    _generateCharmHeader: function() {
+      return (
+        <li className="user-profile__list-header twelve-col">
+          <span className="user-profile__list-col three-col">
+            Name
+          </span>
+          <span className="user-profile__list-col seven-col">
+            Series
+          </span>
+          <span className="user-profile__list-col two-col last-col">
+            Owner
+          </span>
+        </li>);
+    },
+
+    render: function() {
+      var type = this.props.type;
+      if (this.state.loadingEntities) {
+        return (
+          <div className="user-profile__{type}-list twelve-col">
```

Comment:
This will need to be something like ={\`user-profile__${type}-list twelve-col\`}.